parse/grammar/KuneiformParser.g4

```diff
@@ -347,22 +354,29 @@ procedure_block:
     proc_statement*
 ;
 
+// https://docs.kwil.com/docs/kuneiform/operators
 procedure_expr:
-    literal type_cast?                                                                          # literal_procedure_expr
+    // highest precedence:
+    LPAREN procedure_expr RPAREN type_cast?                                                     # paren_procedure_expr
+    | procedure_expr PERIOD IDENTIFIER type_cast?                                               # field_access_procedure_expr
+    | procedure_expr LBRACKET procedure_expr RBRACKET type_cast?                                # array_access_procedure_expr
+    | <assoc=right> (PLUS|MINUS|EXCL) procedure_expr                                            # unary_procedure_expr
```

Comment:
is `!323` and `!age` valid in Postgres?

Reply:
wait I don't see `!` in the operator table https://www.postgresql.org/docs/16/sql-syntax-lexical.html#SQL-PRECEDENCE-TABLE

Reply:
No but we translate it to `not` when we generate code. Also it is only in procedures.

Reply:
OIC